### PR TITLE
Feat/botonic plugin analytics v4

### PR DESCRIPTION
**Pull request by vanbasten17**
## Description
* Replacing plugin-google-analytics by another one which uses GA4.

## Testing

The pull request...

- has unit tests
- has integration tests
- doesn't need tests because... **[provide a description]**

### CODE REVIEW

File: packages/botonic-plugin-google-analytics-v4/package.json
Comment (asastre) [inline packages/botonic-plugin-google-analytics-v4/package.json:3]: Change: The version for this package should be `0.22.0`, right? The latest one released is the `0.21.0`.

File: packages/botonic-plugin-google-analytics-v4/src/index.ts
Comment (asastre) [inline packages/botonic-plugin-google-analytics-v4/src/index.ts:13]: I suggest to set the `userId` also as a function (`getUserId`) the same way it's done for `getClientId`, to allow the bot to decide where to obtain its value rather than fixing it when initializing the plugin.

Also, its value should be added in the constructor.

File: packages/botonic-plugin-google-analytics-v4/src/index.ts
Comment (asastre) [inline packages/botonic-plugin-google-analytics-v4/src/index.ts:16]: These `getEnv` methods shouldn't be used and get the values directly from `options` (to be set when defining the plugin in the bot).

File: packages/botonic-plugin-google-analytics-v4/src/types.ts
Comment (asastre) [inline packages/botonic-plugin-google-analytics-v4/src/types.ts:13]: The attributes of `apiSecret`, `measurementId` and `getUserId` should be added here.

File: packages/botonic-plugin-google-analytics-v4/src/index.ts
Comment (asastre) [inline packages/botonic-plugin-google-analytics-v4/src/index.ts:52]: Not needed.

File: packages/botonic-plugin-google-analytics-v4/src/types.ts
Comment (asastre) [inline packages/botonic-plugin-google-analytics-v4/src/types.ts:9]: I would use directly the interface `Session` rather than extending it for now.

File: packages/botonic-plugin-google-analytics-v4/src/types.ts
Comment (asastre) [inline packages/botonic-plugin-google-analytics-v4/src/types.ts:25]: Not needed

File: packages/botonic-plugin-google-analytics-v4/README.md
Comment (asastre) [inline packages/botonic-plugin-google-analytics-v4/README.md:24]: This README needs to be updated with the new GA4 parameters, right?

**Review by asastre — CHANGES_REQUESTED** [review]
I've added some comments/observations/changes to review before approving the PR :wink:

File: packages/botonic-plugin-google-analytics-v4/README.md
Comment (vanbasten17) [reply packages/botonic-plugin-google-analytics-v4/README.md]: yes, I have replaced it now! 👌

File: packages/botonic-plugin-google-analytics-v4/package.json
Comment (vanbasten17) [reply packages/botonic-plugin-google-analytics-v4/package.json]: changed 👌

File: packages/botonic-plugin-google-analytics-v4/src/index.ts
Comment (vanbasten17) [reply packages/botonic-plugin-google-analytics-v4/src/index.ts]: removed 👍

File: packages/botonic-plugin-google-analytics-v4/src/index.ts
Comment (vanbasten17) [reply packages/botonic-plugin-google-analytics-v4/src/index.ts]: added, thanks for the feedback 🙌

File: packages/botonic-plugin-google-analytics-v4/README.md
Comment (vanbasten17) [reply packages/botonic-plugin-google-analytics-v4/README.md]: yes! maybe I'll need a helping hand here ✋

Comment (asastre) [reply packages/botonic-plugin-google-analytics-v4/README.md]: @vanbasten17 I've updated the `README` with all the changes regarding the new GA4 tracking, can you take a look to see if I'm missing something and if it looks good to you?

Comment (vanbasten17) [reply packages/botonic-plugin-google-analytics-v4/README.md]: LGTM! 👍  let's publish it once it's approved!

**Review by asastre — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**
